x/slashing/handler.go

```diff
@@ -11,8 +13,10 @@ func NewHandler(k Keeper) sdk.Handler {
 		switch msg := msg.(type) {
 		case MsgUnjail:
 			return handleMsgUnjail(ctx, msg, k)
+
 		default:
-			return sdk.ErrTxDecode("invalid message parse in staking module").Result()
+			errMsg := fmt.Sprintf("unrecognized slashing message type: %T", msg)
+			return sdk.ErrUnknownRequest(errMsg).Result()
```

Comment:
Can we have a test case for this please?

Reply:
This will add a lot of boilerplate code just to test a single path (we need to implement an `sdk.Msg`). We could make the private one from `x/mock` public but that means all the modules will not rely on `x/mock`.

Reply:
Nvm! We can use `sdk.NewTestMsg()`.